zjit/src/hir.rs

```diff
@@ -2005,6 +2005,49 @@ impl Function {
             None => {},
         }
     }
+
+
+    /// Validates the following:
+    /// 1. Basic block jump args match parameter arity.
+    /// 2. Every terminator must be in the last position.
+    /// 3. Every block must have a terminator.
+    fn validate_block_terminators_and_jumps(&self) {
```

Comment:
Let's have this return `Result<(), ValidationError>` so we can stop compilation gracefully